File: app/src/main/java/com/gpaddy/baseandroid/quyen/diginews/fragment/HomeFragment.java
package com.gpaddy.baseandroid.quyen.diginews.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import androidx.fragment.app.Fragment;

import com.gpaddy.baseandroid.quyen.diginews.Popup4Activity;
import com.gpaddy.baseandroid.quyen.diginews.Popup5Activity;
import com.gpaddy.baseandroid.quyen.diginews.PopupActivity;
import com.gpaddy.baseandroid.R;

public class HomeFragment extends Fragment {
    private Button btn_popup1, btn_popup2, btn_popup;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.home_fragment, container, false);
        btn_popup1 = v.findViewById(R.id.btn_popup1);
        btn_popup2 = v.findViewById(R.id.btn_popup2);
        btn_popup = v.findViewById(R.id.btn_popup);
        btn_popup1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(HomeFragment.this.getContext(), Popup4Activity.class);
                startActivity(intent);

            }
        });
        btn_popup2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(HomeFragment.this.getContext(), Popup5Activity.class);
                startActivity(intent);

            }
        });

        btn_popup.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(HomeFragment.this.getContext(), PopupActivity.class);
                startActivity(intent);

            }
        });
        return v;
    }
}

File: app/src/main/java/com/gpaddy/baseandroid/ui/fragment/HomeFragment.kt
package com.gpaddy.baseandroid.ui.fragment

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.lifecycle.Observer
import androidx.recyclerview.widget.LinearLayoutManager
import com.gpaddy.baseandroid.BR
import com.gpaddy.baseandroid.R
import com.gpaddy.baseandroid.base.ui.BaseFragment
import com.gpaddy.baseandroid.base.ui.DataBindingConfig
import com.gpaddy.baseandroid.databinding.HomeFragmentBinding
import com.gpaddy.baseandroid.theu.ttvnpt.timkiem
import com.gpaddy.baseandroid.ui.activity.MainActivity
import com.gpaddy.baseandroid.ui.adapter.NewsAdapter
import com.gpaddy.baseandroid.viewmodel.HomeViewModel


class HomeFragment : BaseFragment() {

    private lateinit var viewModel: HomeViewModel
    override fun initViewModel() {
        viewModel=getFragmentScopeViewModel(HomeViewModel::class.java)
    }

    override fun getDataBindingConfig(): DataBindingConfig {
        return DataBindingConfig(
            R.layout.home_fragment,
            BR.vm,
            viewModel
        )
            .addBindingParam(BR.click, ClickProxy())
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val binding = mBinding as HomeFragmentBinding
        val adapte= NewsAdapter()
        binding.rcvNews.apply {
            layoutManager=LinearLayoutManager(context)
            adapter= adapte
        }
        mDisposable.addAll(viewModel.getNewsHome())
        viewModel.newsData.observe(viewLifecycleOwner, Observer {
            adapte.submitList(it.items)
        })
        binding.button
    }

    inner class ClickProxy {
        fun openAccount() {
            val dir= HomeFragmentDirections.actionHomeFragmentToAccountFragment()
            nav().navigate(dir)
        }

        fun openSearch(){
            val intent = Intent( context, timkiem::class.java)
            startActivity(intent)
        }
    }
}